crates/app/src/builder.rs

```diff
@@ -3,32 +3,63 @@
 
 use super::{handle, RepoStore, TagStore, TreeStore};
 
+use drawbridge_store::Storage;
+
+use std::error::Error;
+use std::ffi::OsStr;
+use std::fs::{create_dir, File};
+use std::path::Path;
 use std::sync::Arc;
 
 use axum::handler::Handler;
 use axum::routing::IntoMakeService;
 use axum::{Extension, Router};
+use cap_async_std::fs::Dir;
+use tokio::sync::RwLock;
+
+pub struct Builder<'a> {
+    data_directory: &'a Path,
+}
 
-#[derive(Default)]
-pub struct Builder;
+impl<'a> Default for Builder<'a> {
+    fn default() -> Self {
+        Self {
+            data_directory: Path::new("drawbridge-data"),
+        }
+    }
+}
 
-impl Builder {
+impl<'a> Builder<'a> {
+    /// Constructs a new [Builder].
     pub fn new() -> Self {
-        Self
+        Default::default()
     }
 
-    // TODO: Add configuration functionality
+    /// The directory to be used for all data storage and configuration.
+    pub fn data_directory<S: AsRef<OsStr> + ?Sized>(mut self, path: &'a S) -> Self {
+        self.data_directory = Path::new(path);
+        self
+    }
 
     /// Builds the application and returns Drawbridge instance as a [tower::MakeService].
-    pub fn build(self) -> IntoMakeService<Router> {
-        let repos: Arc<RepoStore> = Default::default();
-        let tags: Arc<TagStore> = Default::default();
-        let trees: Arc<TreeStore> = Default::default();
-        Router::new()
+    pub fn build(self) -> Result<IntoMakeService<Router>, Box<dyn Error>> {
+        let path = Path::new(&self.data_directory);
+
+        if !path.exists() {
+            create_dir(path)?;
```

Comment:
Again, this is potentially *very* sensitive data and we should do the least amount of assumptions. IMO we should just fail in the path does not exist.
We can later add a flag for creating the specified path, if it does not exist, just like `mkdir -p` would do.